### ISSUE TITLE
ssh DOCKER_HOST hangs -> leads to connection time outs

### ISSUE THREAD

**Issue by MLEI-AERO** [docker/compose#8682]
## Description of the issue
When I set up a remote connection by DOCKER_HOST=ssh://... or using a context remote, I am at least 1 time able to run `docker-compose up --build --force-recreate`. As it happens one container exits with a error code, so I have to quit by pressing Ctrl+C. This gives me

`proxy exited with code 1
^CGracefully stopping... (press Ctrl+C again to force)
WARN[0342] commandConn.CloseWrite: commandconn: failed to wait: signal: killed
WARN[0342] commandConn.CloseWrite: commandconn: failed to wait: signal: killed
WARN[0342] commandConn.CloseWrite: commandconn: failed to wait: signal: killed
WARN[0342] commandConn.CloseWrite: commandconn: failed to wait: signal: killed
[+] Running 0/3
[+] Running 3/3bportal  Stopping                                                                                                                                            1.3s
 ⠿ Container webportal  Stopped                                                                                                                                             1.5s 
 ⠿ Container proxy      Stopped                                                                                                                                             1.3s 
 ⠿ Container bootstrap  Stopped                                                                                                                                             2.3s
WARN[0345] commandConn.CloseWrite: commandconn: failed to wait: signal: killed
canceled`

Afterwards I am not able to start again, even a ssh connection directly by `ssh -i key user@ip` is mostly not possible (not always).

## Context information (for bug reports)

- [ ] Using Compose V2 `docker compose ...`
- [x] Using Compose V1 `docker-compose ...`

**Output of `docker(-)compose version`**
```
Docker Compose version v2.0.0-rc.3
```

**Output of `docker version`**
```
Client: Docker Engine - Community
 Cloud integration: 1.0.17
 Version:           20.10.8
 API version:       1.41
 Go version:        go1.16.6
 Git commit:        3967b7d
 Built:             Fri Jul 30 19:54:02 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.8
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.6
  Git commit:       75249d8
  Built:            Fri Jul 30 19:52:00 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.9
  GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
 runc:
  Version:          1.0.1
  GitCommit:        v1.0.1-0-g4144b63
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

## Steps to reproduce the issue

1. DOCKER_HOST=ssh://user@ip
2. eval "$(ssh-agent)"
3. ssh-add ~/.ssh/key_rsa
4. docker-compse up --build --force-recreate
5. Ctrl+C
6. docker-compse up --build --force-recreate

### Observed result
`WARN[0001] commandConn.CloseWrite: commandconn: failed to wait: signal: killed 
[+] Building 0.0s (0/0)
failed to get status: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing command [ssh -l user -- x.x.x.x docker system dial-stdio] has exited with exit status 255, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=ssh: connect to host x.x.x.x port 22: Connection timed out\r\n"`

### Expected result
Work everytime, I don't see this problem on local.

## Additional information
I am using Win10 with WSL Ubuntu 20.04.3 LTS, remote CentOS7 and CentOS8, Visual Studio Code 1.60.2

**Comment by ndeloof** [comment]
duplicates https://github.com/docker/compose/issues/8544

**Comment by MLEI-AERO** [comment]
I don't think this issue is a duplicate of #8544. May the warning message, anyway after the error I am not able to start anything, neither `docker ps`. Further sometimes a system reboot helps, mostly not.
